FeedSX/screens/createPost/index.tsx

```diff
@@ -0,0 +1,549 @@
+      }
+    }
+  };
+
+  // function removes the selected documents
+  const removeDocAttachment = (index: number) => {
+    let newDocAttachments = [...formattedDocAttachments];
+    if (formattedDocAttachments.length === 1) {
+      setFormattedDocAttachments([]);
+      setShowOptions(true);
+    } else {
+      newDocAttachments.splice(index, 1);
+      setFormattedDocAttachments(newDocAttachments);
+    }
+  };
+
+  // function removes multiple images/videos selected
+  const removeMediaAttachment = (index: number) => {
+    let newMediaAttachments = [...formattedMediaAttachments];
+    newMediaAttachments.splice(index, 1);
+    setFormattedMediaAttachments(newMediaAttachments);
+  };
+
+  // function removes single image and video selected
+  const removeSingleAttachment = () => {
+    setFormattedMediaAttachments([]);
+    setShowOptions(true);
+  };
+
+  // function finds the valid links
+  const detectLinks = (text: string) => {
+    const regex = /\b(?:https?:\/\/)?(?:[\w.]+\.\w+)(?:(?<=\\n)|\b)/;
+    const links = text.match(regex);
+
+    if (!showPreview && links?.length === 1) {
+      const decodeUrlResponse = dispatch(
+        getDecodedUrl(
+          DecodeURLRequest.builder().setURL(`${links}`).build(),
+        ) as any,
+      );
+      decodeUrlResponse.then((res: any) => {
+        convertLinkData([res.og_tags]);
+      });
+      setShowPreview(true);
+    } else if (links?.length !== 1) {
+      setFormattedLinkAttachments([]);
+      setShowPreview(false);
+    }
+    return links ? links : [];
+  };
+
+  let allAttachment = [
+    ...formattedMediaAttachments,
+    ...formattedDocAttachments,
+  ];
+
+  return (
+    <SafeAreaView style={{flex: 1, backgroundColor: '#fff'}}>
+      {/* screen header section*/}
+      <LMHeader
+        showBackArrow
+        onBackPress={() => NavigationService.navigate('UniversalFeed')}
+        heading="Create a Post"
+        rightComponent={
+          <TouchableOpacity
+            disabled={
+              allAttachment?.length > 0 ||
+              formattedLinkAttachments?.length > 0 ||
+              postContentText != ''
+                ? false
+                : true
+            }
+            style={{
+              opacity:
+                allAttachment?.length > 0 ||
+                formattedLinkAttachments?.length > 0 ||
+                postContentText != ''
+                  ? 1
+                  : 0.5,
+            }}
+            onPress={() => {
+              dispatch(
+                setUploadAttachments({
+                  mediaAttachmentData: allAttachment,
+                  linkAttachmentData: formattedLinkAttachments,
+                  postContentData: postContentText,
+                }) as any,
+              );
+              NavigationService.goBack();
+            }}>
+            <Text>POST</Text>
+          </TouchableOpacity>
+        }
+      />
+      <ScrollView>
+        {/* user profile section */}
+        <View
+          style={{
```

Comment:
Can you put these styles in `styles.ts` file?

Reply:
resolved